stdlib/typing.pyi

```diff
@@ -132,9 +132,6 @@ if sys.version_info >= (3, 12):
 ContextManager = AbstractContextManager
 AsyncContextManager = AbstractAsyncContextManager
 
-# This itself is only available during type checking
-def type_check_only(func_or_cls: _F) -> _F: ...
```

Comment:
Why were these moved? This is legal in stubs, even if it might be better style to move the function after the definition of `_F`.

Reply:
(If it does get added to a spec somewhere or we think there _is_ a good reason to support _arbitrary_ forward references in stubs, I'm personally very open to adding support for it at Ruff.)

Reply:
That makes sense in general; the second change in this PR (`connect = Connection` before `Connection` is defined) is definitely some very odd code.

I'm not sure why Ruff has an issue with the definition of `def type_check_only`, though. `_F` is used only in an annotation; why does the definition of `type_check_only` need to be after the definition of `_F`?

Reply:
The issue is that the TypeVar `_F` is bound to `Callable` before `Callable` is defined

Reply:
Ah, I guess that's the kind of case we only encounter in typeshed :)

Reply:
Yup! 😄